### PR TITLE
reworked sqs_queue

### LINKED ISSUE #41260: Get SQS url as return value
##### SUMMARY
Currently there is no way to get the SQS url of an existing queue or a newly created queue either in ansible 2.5 or 2.6

##### ISSUE TYPE
 - Feature Idea

##### COMPONENT NAME
`lib/ansible/modules/cloud/amazon/sqs_queue.py`

##### ANSIBLE VERSION
```
ansible 2.5.4
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/abhi/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/dist-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.15rc1 (default, Apr 15 2018, 21:51:34) [GCC 7.3.0]
```

##### CONFIGURATION
abhi@:~$ ansible-config dump --only-changed
abhi@:~$ 

##### OS / ENVIRONMENT
Running from - ubuntu 18.04
Managing - RHEL 7

##### STEPS TO REPRODUCE
It is pretty simple to reproduce, create a new sqs queue and there is just no way to get the queue_url.

```yaml
- name: Create SQS queue
  aws_sqs_queue:
    aws_access_key: "{{ profile_aws_access_key }}"
    aws_secret_key: "{{ profile_aws_secret_key }}"
    security_token: "{{ profile_security_token }}"
    name: "sqs_queue"
    region: "{{ default_region }}"
    default_visibility_timeout: 300
    state: present
  register: sqs_queue

- debug: var=sqs_queue
```

##### EXPECTED RESULTS
I am expecting to have a queue_url value in the returned parameters.

##### ACTUAL RESULTS

```
ok: [localhost] => {
    "ses_queue": {
        "changed": false, 
        "default_visibility_timeout": "300", 
        "delivery_delay": "0", 
        "failed": false, 
        "maximum_message_size": "262144", 
        "message_retention_period": "345600", 
        "name": "sqs_queue", 
        "policy": null, 
        "queue_arn": ERASED, 
        "receive_message_wait_time": "0", 
        "redrive_policy": null, 
        "region": "us-east-1"
    }
}
```

### PR DESCRIPTION
**Pull request by sbj-ss**
##### SUMMARY
Fully reworked `sqs_queue` module.
- Moved to AnsibleAWSModule
- Compacted code
- Added FIFO queues, tagging and KMS support
- Added ability to remove policies.

Fixes #41260, #43775
##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
`lib.ansible.modules.cloud.amazon.sqs_queue`

##### ADDITIONAL INFORMATION
I'm aware of #34526, but the change request looks stuck and that implementation contains some bugs.

### CODE REVIEW

**Comment by mattclay** [comment]
CI failure in integration tests: https://app.shippable.com/github/ansible/ansible/runs/101235/68/tests

**Comment by sbj-ss** [comment]
@mattclay Please see this: https://github.com/ansible/ansible/pull/40110/commits/6e44bd6b8c1152edaa4ab6c5d483fbd9e5ab257c#diff-727e795eecd556311091aadde40702abR58
Tests are failing because they're expecting wrong result type. I can open another PR to fix tests :)

**Comment by mattclay** [comment]
@sbj-ss If the tests can be fixed independently of the changes in this PR and still pass, then a separate PR to fix would be good. Otherwise they'll need to be fixed as part of this PR, since we can't merge changes that result in failing tests.

**Comment by sbj-ss** [comment]
@mattclay will tests included in PR be used to test the PR itself?

**Comment by mattclay** [comment]
@sbj-ss Yes

**Comment by samicemalone** [comment]
Could the queue URL be returned to resolve #41260 ?
Also is it possible to return the queue ARN (and URL) in check mode (if the queue exists)?

**Comment by sbj-ss** [comment]
@samicemalone Yes, will add them today.

**Comment by sbj-ss** [comment]
@mattclay @samicemalone done.

**Comment by mente** [comment]
Any chances to move this forward? What is missing here?

File: lib/ansible/modules/cloud/amazon/sqs_queue.py [hunk 1]
```diff
@@ -130,178 +201,178 @@
...
-    if not HAS_BOTO:
-        module.fail_json(msg='boto required for this module')
-
-    region, ec2_url, aws_connect_params = get_aws_connection_info(module)
-    if not region:
-        module.fail_json(msg='region must be specified')
+def update_tags(client, queue_url, module):
+    new_tags = module.params.get('tags')
+    purge_tags = module.params.get('purge_tags')
+    if new_tags is None:
```
Comment (tremble) [inline lib/ansible/modules/cloud/amazon/sqs_queue.py:339]: This logic seems ignore the use case of removing all tags.

File: lib/ansible/modules/cloud/amazon/sqs_queue.py [hunk 2]
```diff
@@ -23,48 +23,86 @@
...
     description:
       - Name of the queue.
     required: true
-  default_visibility_timeout:
+  queue_type:
+    description:
+      - Standard or FIFO queue.
+    choices: ['standard', 'fifo']
+    default: 'standard'
+    version_added: "2.8"
```
Comment (tremble) [inline lib/ansible/modules/cloud/amazon/sqs_queue.py:44]: This will need updating

File: lib/ansible/modules/cloud/amazon/sqs_queue.py [hunk 3]
```diff
@@ -23,48 +23,86 @@
...
     choices: ['present', 'absent']
     default: 'present'
   name:
     description:
       - Name of the queue.
     required: true
-  default_visibility_timeout:
+  queue_type:
+    description:
+      - Standard or FIFO queue.
```
Comment (tremble) [inline lib/ansible/modules/cloud/amazon/sqs_queue.py:41]: This should probably mention that the queue_type can only be selected at creation time and is *ignored* when making other changes

**Review by tremble — CHANGES_REQUESTED** [review]
@sbj-ss Sorry you've not heard anything for so long.

Since you're adding new features to a module marked as "stableinterface" please add tests for the additional features.

I've had a quick run through the code nothing else stands out.

**Comment by citrix-ericgo** [comment]
Any update on this? Looks like a merge conflict.

**Comment by sbj-ss** [comment]
Sorry, I'm now very far from ansible on my new job. If someone wants to push this forward - you're really welcome.

**Pull request closed without merging**